### PR TITLE
WIP: add gradlew build workflow

### LINKED ISSUE #4: Setup CI/CD pipeline
**Is your feature request related to a problem? Please describe.**
We want to have CI/CD in this project using one of the following:
- Github Actions
- TravisCI
- CircleCI
- or any other

**Describe the solution you'd like**
Have a configurable yaml file in project root to build, test and publish the projects.

### PR DESCRIPTION
**Pull request by pedromsmoreira**
This PR partially closes #4 

add build for backend-api 

We can create more issues, one for each project.

### CODE REVIEW

**Comment by joao-r-reis** [comment]
Merge/rebase this branch so that we can test changes that were merged in #11 and #14

**Comment by joao-r-reis** [comment]
This addresses #4 right? you can add something like "fix #4" to the PR description so that it closes it automatically after the merge.

File: backend-api/build.gradle [hunk 1]
```diff
@@ -17,6 +17,9 @@ version '1.0-SNAPSHOT'
 
 sourceCompatibility = 1.11
 
+bootJar {
+    mainClassName = 'pms.PetMgtSystemApplication'
+}
```
Comment (joao-r-reis) [inline backend-api/build.gradle]: hmm this shouldn't be needed AFAIK

File: .github/workflows/gradle.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: Set up JDK 11
+      uses: actions/setup-java@v1
+      with:
+        java-version: 11
+    - name: Grant rights build.gradle
+      run: chmod +x ./backend-api/build.gradle
```
Comment (joao-r-reis) [inline .github/workflows/gradle.yml]: are you sure this needs `+x`?

Comment (pedromsmoreira) [reply .github/workflows/gradle.yml]: no need, just an experiment. It isn't an executable. I will remove it.

File: backend-api/build.gradle (hunk 1 above)
Comment (pedromsmoreira) [reply backend-api/build.gradle]: We can remove it, the build result was the same with or without it.

Comment (joao-r-reis) [reply backend-api/build.gradle]: https://docs.spring.io/spring-boot/docs/2.2.0.RELEASE/gradle-plugin/reference/html/#packaging-executable-jars

> Executable jars can be built using the bootJar task. The task is automatically created when the java plugin is applied and is an instance of BootJar. The assemble task is automatically configured to depend upon the bootJar task so running assemble (or build) will also run the bootJar task.

We are using the `java` plugin so `gradle build` should be generating the jar file already.

File: .github/workflows/gradle.yml (hunk 2 above)
Comment (pedromsmoreira) [reply .github/workflows/gradle.yml]: removed

**Comment by joao-r-reis** [comment]
Pushed some changes to gradle action so that it uses https://github.com/marketplace/actions/gradle-command and runs tests

**Comment by pedromsmoreira** [comment]
I will close this PR since #17 seems to be worth it in the future when github actions gets the caching feature.

**Pull request closed without merging**